src/Hydrodynamics/EnergyRadws.cpp

```diff
@@ -207,22 +208,21 @@ EnergyRadws<ndim,ParticleType>::~EnergyRadws()
 template <int ndim, template <int> class ParticleType>
 void EnergyRadws<ndim,ParticleType>::EnergyIntegration
  (const int n,                         ///< [in] Integer time in block time struct
-  const int Npart,                     ///< [in] Number of particles
   const FLOAT t,                       ///< [in] Current simulation time
   const FLOAT timestep,                ///< [in] Base timestep value
-  Particle<ndim>* part_gen)            ///< [inout] Pointer to SPH particle array
+  Hydrodynamics<ndim>* hydro)
 {
   int i;                               // Particle counter
   FLOAT dt;                            // Timestep since start of step
-  ParticleType<ndim>* partdata = static_cast<ParticleType<ndim>* > (part_gen);
+  ParticleType<ndim>* partdata = hydro->template GetParticleArray<ParticleType>();
```

Comment:
Nothing.  For some reason the standar requires template keyword when calling a template function with a template parameter that is itself a template parameter.

The motivation here is that GetParticleArray can now do some sanity checking on the requested type.